cmd/lncli/commands.go

```diff
@@ -4016,3 +4030,255 @@ func restoreChanBackup(ctx *cli.Context) error {
 
 	return nil
 }
+
+var newMacaroonCommand = cli.Command{
+	Name:     "newmacaroon",
+	Category: "Macaroons",
+	Usage: "Bakes a new macaroon with the provided list of " +
+		"permissions and restrictions",
+	ArgsUsage: "--permission=entity/action [--format= | --save_to=] " +
+		"[--timeout=] [--ip_address=]",
+	Description: `
+	Create a new macaroon that grants the provided permissions and
+	optionally adds restrictions (timeout, IP address) to it.
+
+	The new macaroon can be shown on command line either in JSON
+	(--format=json) or hex (--format=hex) serialized format.
+	Or it can be saved directly to a file using the --save_to argument.
+
+	A permission is a tuple of an entity and an action.
+	The following entities are available in lnd:
+		* onchain
+		* offchain
+		* address
+		* message
+		* peers
+		* info
+		* invoices
+		* signer
+		* macaroon
+
+	The action can be either 'read', 'write' or 'generate'.
+	Multiple operations can be added, for example:
+
+	--permission=info/read --permission=invoices/write --permission=...
+	`,
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name: "format",
+			Usage: "the format to serialize and display the " +
+				"macaroon in on the command line, must be " +
+				"either 'json' or 'hex' (default 'json')",
+		},
+		cli.StringFlag{
+			Name: "save_to",
+			Usage: "save the created macaroon to this file " +
+				"using the default binary format",
+		},
+		cli.Uint64Flag{
+			Name: "timeout",
+			Usage: "the number of seconds the macaroon will be " +
+				"valid before it times out",
+		},
+		cli.StringFlag{
+			Name:  "ip_address",
```

Comment:
Perhaps leave this and `timeout` out since they can already be specified with the `lncli` flags?

Reply:
The IP address and timeout are already provided by `lncli`, yes. But the user can never get access to those attenuated macaroons since they are only ever used to be sent over the wire. So I think it's still useful to be able to mint and directly restrict a macaroon with this command. Especially when we can do it client side with minimal effort.